velox/core/PlanNode.h

```diff
@@ -4713,31 +4713,75 @@ class MarkDistinctNode : public PlanNode {
   const RowTypePtr outputType_;
 };
 
-/// Optimized version of a WindowNode for a single row_number function with a
-/// limit over sorted partitions.
-/// The output of this node contains all input columns followed by an optional
+/// Optimized version of a WindowNode for a single row_number, rank or
+/// dense_rank function with a limit over sorted partitions. The output of this
+/// node contains all input columns followed by an optional
 /// 'rowNumberColumnName' BIGINT column.
+/// TODO: This node will be renamed to TopNRank once all the support for
+/// handling rank and dense_rank is committed to Velox.
 class TopNRowNumberNode : public PlanNode {
```

Comment:
Do we need to have a TopNWindowNode as base class and extends it by TopNRowNumberNode, TopNRankNode, xxx? Not sure for that. CC @rui-mo

Reply:
I think it would be good if we could leverage much of the TopNRank node to handle several functions. Also, the row_number, rank, and dense_rank appear to be more function-level than operator-level distinctions.

Reply:
Do you have suggest for the node name? And in that case, do we need 3 functions for PlanBuilder? https://github.com/facebookincubator/velox/pull/13248/files#r2134892580

Reply:
Your comment says `TODO: This node will be renamed to TopNRank once all the support for`, so I try to find a reasonable new name.
if we have only one node, I would prefer PlanBuilder has 1 function.

Reply:
@jinchengchenghh : I intend to rename TopNRowNumberNode to TopNRank in the future as that is more representative of the functionality.

So we generally don't want the user to type internal PlanNode enum values (as would be needed for the function here) if possible. So one of the options was to take a string "row_number()", "rank()" or "dense_rank()"" function name in the single

Reply:
I don't have strong opinion for it. One node for one function in PlanBuilder. Maybe the name TopNRowNode is more reasonable.

Reply:
@jinchengchenghh : Trino uses the name TopNRankingNode for this PlanNode. Presto uses TopNRowNumberNode (but it doesn't have rank, dense_rank). Does Spark have an equivalent node and what is it called there ?

I feel we should choose a name consistent with the engines.

Reply:
Spark use WindowGroupLimitExec name https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowGroupLimitExec.scala#L44